### ISSUE TITLE
decimal.DivisionByZero: [<class 'decimal.DivisionByZero'>]

### ISSUE THREAD

**Issue by kristalikas** [nasa/HyperCP#253]
![image](https://github.com/user-attachments/assets/f1997e24-6086-4ffe-ba25-9709530fb6db)

![Screenshot 2024-10-14 at 14 34 24](https://github.com/user-attachments/assets/94c80cf9-991b-4cf2-bf0d-0036f63eeb7a)

**Comment by oceancolorcoder** [comment]
@kristalikas, thanks for posting this. Could you send me the L1BQC HDF file you are processing here for us to test?

**Comment by kristalikas** [comment]
@oceancolorcoder Thanks! I sent the L1BQC HDF file.

**Comment by oceancolorcoder** [comment]
@kristalikas files received. @ARamsay17 could you please take a look at this? Files are in Data/Tartu/253

**Comment by ARamsay17** [comment]
Hi @kristalikas it seems the issue was caused by a single pixel having a value of 0 in it's non linearity calibration. I have added a check to make sure this cannot create any issues in the uncertainty propagation.

Also I found NaN in the LI raw data. This isn't a serious issue but does of course create conflicts in the uncertainty propagation code. @oceancolorcoder I think this is something that we should discuss further with a screening process being added to L1A or L1AQC.
